### PR TITLE
API provides client with list of Fikas

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/180227536

### User Story
```
As an API
In order to give the client an overview of the schedule
I would like to provide with a fika list
```
### Changes Proposed
- Generate Fika Controller
- Generate route
- Create `index` action in the `fikas_controller`
- Create `:participant` for association in Fika model
- `:participant` tests
- Create Fika model with `:start_date`, `:end_date` &`:participants`
- Add `active_model_serializers` gem
- Configure serializer for fikas
- Test for the sad path

## Screenshots
image

### CODE REVIEW

**Review by dorianbuck — APPROVED** [review]
LGTM

File: spec/models/fika_goer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    it { is_expected.to have_db_column :start_date }
+    it { is_expected.to have_db_column :management }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :email }
+    it { is_expected.to validate_presence_of :start_date }
+    it { should allow_value(true).for(:management) }
+    it { should allow_value(false).for(:management) }
```
Comment (Fraseer) [inline spec/models/fika_goer_spec.rb]: gold star

File: spec/requests/api/fikas/fika_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  describe 'when there are no fikas in the databse' do
+    before do
+      get '/api/fikas'
+    end
+
+    it { is_expected.to have_http_status 404 }
+
+    it 'is expected to return with a message indicating there are no articles' do
+      expect(response_json['message']).to eq 'There are no fikas in the database'
+    end
```
Comment (Fraseer) [inline spec/requests/api/fikas/fika_index_spec.rb:41]: What will happen if only 1 fika goer is on the fika?

**Review by Fraseer — COMMENTED** [review]
Looks good, just one comment

File: spec/requests/api/fikas/fika_index_spec.rb (hunk 2 above)
Comment (maxarvid) [reply spec/requests/api/fikas/fika_index_spec.rb:41]: As it's currently implemented, the index action would return a fika with a single participant. We have a [story coming up](https://www.pivotaltracker.com/story/show/180227722) that specifies two participants, maybe we can address this then

File: spec/models/fika_goer_spec.rb (hunk 1 above)
Comment (maxarvid) [reply spec/models/fika_goer_spec.rb]: If of interest, the recommendation from the [Shoulda-Matchers docs](https://matchers.shoulda.io/docs/v5.0.0/Shoulda/Matchers/ActiveModel.html#validate_inclusion_of-instance_method) advises against testing for the inclusion of booleans:
```
We discourage using validate_inclusion_of with boolean columns. In fact, there is never a case where a boolean column will be anything but true, false, or nil, as ActiveRecord will type-cast an incoming value to one of these three values. That means there isn't any way we can refute this logic in a test.
```

File: app/models/fika.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Fika < ApplicationRecord
+  validates_presence_of :start_date, :end_date
+
+  has_and_belongs_to_many :fika_goers, join_table: 'fikas_fika_goers'
```
Comment (oliverochman) [inline app/models/fika.rb]: Everything looks great except this hahaha

Comment (oliverochman) [reply app/models/fika.rb]: but I dont have a better solution....

Comment (oliverochman) [reply app/models/fika.rb]: atm

Comment (maxarvid) [reply app/models/fika.rb]: Hopefully solved by changing `fika_goer` to `participant`

File: db/migrate/20211109084738_create_fikas.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateFikas < ActiveRecord::Migration[6.1]
+  def change
+    create_table :fikas do |t|
+      t.string :start_date
```
Comment (oliverochman) [inline db/migrate/20211109084738_create_fikas.rb]: why not use datetime here?

File: db/migrate/20211109094557_create_participants.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CreateParticipants < ActiveRecord::Migration[6.1]
+  def change
+    create_table :participants do |t|
+      t.string :name
+      t.string :email
+      t.datetime :start_date
```
Comment (oliverochman) [inline db/migrate/20211109094557_create_participants.rb:6]: but use it here?

File: db/migrate/20211109084738_create_fikas.rb (hunk 4 above)
Comment (maxarvid) [reply db/migrate/20211109084738_create_fikas.rb]: Unfounded fear of datetime, fixed now.

File: db/migrate/20211109094557_create_participants.rb (hunk 5 above)
Comment (maxarvid) [reply db/migrate/20211109094557_create_participants.rb:6]: See above, converted the fikas attributes to datetime.

**Pull request merged**